distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1181,7 +1208,44 @@ public void testRetrievalReliablility() throws Exception {
                         Condition.parameter("request", "GetRecordById"),
                         Condition.parameter("id", metacardId));
 
-        // Add CometD notification and activity assertions when DDF-2272 ihas been addressed.
+        expect("Waiting for notifications").within(10, SECONDS)
+                .until(() -> localhostCometDClient.getMessages(NOTIFICATIONS_CHANNEL)
+                        .size() == 3);
+        expect("Waiting for activities").within(10, SECONDS)
+                .until(() -> localhostCometDClient.getMessages(ACTIVITIES_CHANNEL)
+                        .size() == 9);
+
+        List<String> notifications = localhostCometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(3));
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)), filename,
```

Comment:
If the underlying implementation ever changes and it attempts to retry more than three times, this will break.